runtime/include/QuantumDevice.hpp

```diff
@@ -55,310 +57,558 @@ struct QuantumDevice {
     QuantumDevice(QuantumDevice &&) = delete;
     QuantumDevice &operator=(QuantumDevice &&) = delete;
 
-    /**
-     * @brief Allocate a qubit.
-     *
-     * @return `QubitIdType`
-     */
-    virtual auto AllocateQubit() -> QubitIdType = 0;
+    // ----------------------------------------
+    //  QUBIT MANAGEMENT
+    // ----------------------------------------
 
     /**
-     * @brief Allocate a vector of qubits.
+     * @brief Allocate an array of qubits.
+     *
+     * The operation should perform the necessary steps to make the given number of qubits
+     * available in a clean |0> state. At minimum, a device must be able to execute this
+     * call once before any quantum operations have been run. Handling multiple allocation calls
+     * is optional and only useful if the device intends to support dynamic qubit allocations
+     * (that is allocations during quantum program execution).
+     *
+     * While devices may choose not to distinguish between logical and device IDs, having a logical
+     * qubit labeling system can help catch program errors such as addressing a previously released
+     * qubit. An example implementation can be found in `QubitManager.hpp`. Devices are also free
+     * to disable a resource without physically freeing it, allowing for faster dynamic allocations.
+     * The returned IDs must be accepted in addressing operations (like gates) until the quantum
+     * resource is explicitly freed or the device is destroyed.
      *
      * @param num_qubits The number of qubits to allocate.
      *
-     * @return `std::vector<QubitIdType>`
+     * @return `std::vector<QubitIdType>` Array of logical qubit IDs.
```

Comment:
```suggestion
     * @return `std::vector<QubitIdType>` Array of qubit IDs.
```

since we are saying just device qubit IDs is also ok?

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1680/commits/e7c92a7ead42d8ac4d000addedfa0cfa5c69405f